crates/krun-server/src/bin/krun-server.rs

```diff
@@ -0,0 +1,31 @@
+use std::env;
+use std::process::Command;
+use std::thread;
+
+use anyhow::Result;
+use krun_server::cli_options::options;
+use krun_server::server::start_server;
+
+fn main() -> Result<()> {
+    env_logger::init();
+
+    let options = options().run();
+
+    let server_port = env::var("KRUN_SERVER_PORT").unwrap_or("3333".to_string());
+    let server_port: u32 = server_port.parse().unwrap_or(3333);
+
+    thread::spawn(move || {
+        if let Err(e) = start_server(server_port) {
+            println!("Error starting krun server: {e}")
+        }
+    });
+
+    let err = Command::new(&options.command)
+        .args(options.command_args)
+        .status();
+    if let Err(err) = err {
+        println!("Error executing command {}: {}", options.command, err);
+    }
+
+    std::process::exit(0);
```

Comment:
Do we not intend to keep the VM running in the background? To me that seems like it'd be much better for UX, in the sense that running a command via `krun` would be instant if the VM / `krun-server` is already running in the background.

Reply:
I'm imagining a UX like this:

```
krun-server start
krun ...
krun ...
krun-server stop
```

Which makes it kinda analogous to `podman start` / `podman stop`?

Reply:
> Do we not intend to keep the VM running in the background? To me that seems like it'd be much better for UX, in the sense that running a command via krun would be instant if the VM / krun-server is already running in the background.

This is what happens with this PR. If krun is already running, launching the command feels instantaneous.

> I'm imagining a UX like this:
>
> krun-server start
> krun ...
> krun ...
> krun-server stop 
> Which makes it kinda analogous to podman start / podman stop?
>
> (krun can autostart krun-server if it's not already running? Socket activation?)

The UX implemented in this PR is better. It allows users to preserve the current workflow, just running a single command that will tear down the VM when finished (which is the spirit of microVMs anyways), but it also allows users to run multiple commands on the same VM seamlessly. It's even compatible with binfmt (I've tested it this weekend).

Starting the command will always feel instantaneous. The only reason it takes a bit now when VM is not running is due to `dhcpcd` doing stupid things, something I intend to fix in a future PR.

Installing krun as a systemd socket-activated service will imply an implicit, hidden lifetime for the VM, which can be confusing (why is krun still running if I've just closed the app?).